src/components/autoSuggest/index.test.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+import { I18nextProvider } from 'react-i18next';
+import { spy } from 'sinon';
+import AutoSuggest from './index';
+import i18n from '../../i18n';
+
+import * as searchActions from './../search/keyAction';
+
+import routes from '../../constants/routes';
+import keyCodes from './../../constants/keyCodes';
+import mockSearchResults from './searchResults.mock';
+
+describe('AutoSuggest', () => {
+  let wrapper;
+  let props;
+  let results;
+  let submitSearchSpy;
+  let visitAndSaveSearchSpy;
+
+  beforeEach(() => {
+    results = { ...mockSearchResults };
+    props = {
+      t: key => key,
+      history: {
+        push: spy(),
+      },
+      results,
+    };
+
+    submitSearchSpy = spy(AutoSuggest.prototype, 'submitSearch');
+    visitAndSaveSearchSpy = spy(searchActions, 'visitAndSaveSearch');
+    wrapper = mount(<I18nextProvider i18n={i18n}><AutoSuggest {...props} /></I18nextProvider>);
+    wrapper.update();
+  });
+
+  afterEach(() => {
+    submitSearchSpy.restore();
+    visitAndSaveSearchSpy.restore();
+  });
+
+  it('should render a row for each entity found {addresses,delegates,transactions}', () => {
+    expect(wrapper.find('.addresses-result')).to.have.lengthOf(3);
+    expect(wrapper.find('.addresses-header')).to.have.lengthOf(1);
+    expect(wrapper.find('.delegates-result')).to.have.lengthOf(3);
+    expect(wrapper.find('.delegates-header')).to.have.lengthOf(1);
+    expect(wrapper.find('.transactions-result')).to.have.lengthOf(3);
+    expect(wrapper.find('.transactions-header')).to.have.lengthOf(1);
```

Comment:
Please use [chai-enzyme exactly](https://github.com/producthunt/chai-enzyme#exactly)
`expect(wrapper).to.have.exactly(3).descendants('.addresses-result');` 
here and elsewhere in this file.

Reply:
done